### ISSUE TITLE
extraSelections() failing

### ISSUE THREAD

**Issue by phoe** [Shinmera/qtools#33]
via https://github.com/commonqt/commonqt/issues/37

Using recent commonqt, `(setf (q+:extra-selections text-edit) selections)` now fails with:

```common-lisp
The alien function "sw_qlist_extraselection_new" is undefined.
   [Condition of type SB-KERNEL::UNDEFINED-ALIEN-FUNCTION-ERROR]

Restarts:
 0: [ABORT] Abort smoke callback
 1: [RETRY] Retry SLIME REPL evaluation request.
 2: [*ABORT] Return to SLIME's top level.
 3: [ABORT] abort thread (#<THREAD "repl-thread" RUNNING {1002938603}>)

Backtrace:
  0: ("undefined function")
      [No Locals]
  1: ((LABELS QT::|const QList<QTextEdit::ExtraSelection>&-MARSHALLER| :IN "/home/phoe/.roswell/lisp/quicklisp/dists/quicklisp/software/commonqt-20180131-git/qlist.lisp") NIL #<CLOSURE (LABELS QT::MARSHAL-..
      Locals:
        QT::CONT = #<CLOSURE (LABELS QT::MARSHAL-AROUND-INNER :IN QT::MARSHALLER) {100C42248B}>
        QT::VALUE = NIL
  2: ((LABELS QT::MARSHAL-AROUND-OUTER :IN QT::MARSHALLER) NIL #.(SB-SYS:INT-SAP #X7F420C2DB090) 1 #<CLOSURE (LAMBDA NIL :IN QT::ARGSTEP-MARSHALLER) {100C42245B}>)
      Locals:
        QT::AROUND-THUNK = #<FUNCTION (LABELS QT::|const QList<QTextEdit::ExtraSelection>&-MARSHALLER| :IN "/home/phoe/.roswell/lisp/quicklisp/dists/quicklisp/software/commonqt-20180131-git/qlist.lisp") {2255013B}>
        QT::CONT = #<CLOSURE (LAMBDA () :IN QT::ARGSTEP-MARSHALLER) {100C42245B}>
        QT::I = 1
        QT::SET-THUNK = #<FUNCTION (LAMBDA (QT::VAL QT::STACK QT::I) :IN QT::SET-THUNK) {225928FB}>
        QT::STACK = #.(SB-SYS:INT-SAP #X7F420C2DB090)
        QT::VALUE = NIL
  3: ((LAMBDA (QT::STACK QT::ARGLIST QT::FINAL-CONT) :IN QT::ARGSTEP-MARSHALLER) #.(SB-SYS:INT-SAP #X7F420C2DB090) (NIL) #<CLOSURE (LAMBDA (QT::STACK) :IN QT::%%CALL) {100C42241B}>)
      Locals:
        QT::ARGLIST = (NIL)
        QT::FINAL-CONT = #<CLOSURE (LAMBDA (QT::STACK) :IN QT::%%CALL) {100C42241B}>
        QT::I = 1
        QT::MARSHAL-THUNK = #<CLOSURE (LABELS QT::MARSHAL-AROUND-OUTER :IN QT::MARSHALLER) {1018E5112B}>
        QT::NEXT-THUNK = #<FUNCTION (LAMBDA (QT::STACK QT::ARGLIST QT::FINAL-CONT) :IN QT::ARGSTEP-MARSHALLER) {2257D55B}>
        QT::STACK = #.(SB-SYS:INT-SAP #X7F420C2DB090)
  4: ((LABELS QT::ARGLIST-MARSHALLER :IN QT::ARGLIST-MARSHALLER) (NIL) #<CLOSURE (LAMBDA (QT::STACK) :IN QT::%%CALL) {100C42241B}>)
      Locals:
        QT::ARGLIST = (NIL)
        QT::FINAL-CONT = #<CLOSURE (LAMBDA (QT::STACK) :IN QT::%%CALL) {100C42241B}>
        QT::N = 2
        #:SIZE0 = 16
        QT::STACK = #.(SB-SYS:INT-SAP #X7F420C2DB090)
        QT::THUNK = #<CLOSURE (LAMBDA (QT::STACK QT::ARGLIST QT::FINAL-CONT) :IN QT::ARGSTEP-MARSHALLER) {1018E24B4B}>
  5: ((LAMBDA (QT::ACTUAL-INSTANCE QT::ARGS) :IN QT::RESOLVE-CALL) #<SPELLCHECKED-TEXT-EDIT QTextEdit 0x7F420C144110> (NIL))
      Locals:
        QT::<FROM> = 25092
        QT::<TO> = 25092
        QT::ACTUAL-INSTANCE = #<SPELLCHECKED-TEXT-EDIT QTextEdit 0x7F420C144110>
        QT::ARGLIST-MARSHALLER = #<CLOSURE (LABELS QT::ARGLIST-MARSHALLER :IN QT::ARGLIST-MARSHALLER) {1018D9AA8B}>
        QT::ARGLIST-MARSHALLER#1 = #<CLOSURE (LABELS QT::ARGLIST-MARSHALLER :IN QT::ARGLIST-MARSHALLER) {1018D9AA8B}>
        QT::ARGS = (NIL)
        QT::ARGS#1 = (NIL)
        QT::CASTED-INSTANCE-POINTER = #.(SB-SYS:INT-SAP #X7F420C144110)
        QT::CASTFN = #.(SB-SYS:INT-SAP #X7F42112781A0)
        QT::CLASSFN = #.(SB-SYS:INT-SAP #X7F42113DADD0)
        QT::CLASSFN#1 = #.(SB-SYS:INT-SAP #X7F42113DADD0)
        QT::METHOD-INDEX = 63
        QT::METHOD-INDEX#1 = 63
        QT::RETURN-VALUE-FUNCTION = #<CLOSURE (LAMBDA (&REST SB-IMPL::ARGUMENTS) :IN CONSTANTLY) {1018D9AAAB}>
        QT::RETURN-VALUE-FUNCTION#1 = #<CLOSURE (LAMBDA (&REST SB-IMPL::ARGUMENTS) :IN CONSTANTLY) {1018D9AAAB}>
```

Also it seems that finalization does not yet work on these objects:

```common-lisp
No applicable method ~QTextEdit::ExtraSelection found on #<Qt::QTextEdit::ExtraSelection  {1001E25373}> with arguments NIL
   [Condition of type SIMPLE-ERROR]

Restarts:
 0: [ABORT] Abort smoke callback
 1: [RETRY] Retry SLIME REPL evaluation request.
 2: [*ABORT] Return to SLIME's top level.
 3: [ABORT] abort thread (#<THREAD "repl-thread" RUNNING {1002938603}>)

Backtrace:
  0: (QT::RESOLVE-CALL NIL #<Qt::QTextEdit::ExtraSelection  {1001E25373}> "~QTextEdit::ExtraSelection" NIL NIL)
      Locals:
        ALLOW-OVERRIDE-P = NIL
        ARGS = NIL
        FIX-TYPES = NIL
        INSTANCE = #<Qt::QTextEdit::ExtraSelection  {1001E25373}>
        METHOD = "~QTextEdit::ExtraSelection"
        METHOD#1 = NIL
        NAME = "~QTextEdit::ExtraSelection"
  1: (MAYBE-DELETE-QOBJECT #<Qt::QTextEdit::ExtraSelection  {1001E25373}>)
      Locals:
        QT::DTOR = "~QTextEdit::ExtraSelection"
        #:G10 = 25156
        #:G11 = :INSTANCE
        #:HASH-13 = 5
        QT::INSTANCE = #<Qt::QTextEdit::ExtraSelection  {1001E25373}>
        #:INSTANCE-EXTRA-SIG7 = :INSTANCE
        #:INSTANCE-QCLASS6 = 25156
        #:INSTANCE8 = #<Qt::QTextEdit::ExtraSelection  {1001E25373}>
        OBJECT = #<Qt::QTextEdit::ExtraSelection  {1001E25373}>
        QT::OBJECT#1 = #<Qt::QTextEdit::ExtraSelection  {1001E25373}>
        #:PLACES-12 = #(NIL NIL NIL NIL (#<CLOSURE (LAMBDA # :IN QT::RESOLVE-CALL) {1015D7742B}> 25092 #:G1075) (#<CLOSURE (LAMBDA # :IN QT::RESOLVE-CALL) {1001ABC02B}> 8384 :INSTANCE) ...)
        #:PREVIOUS-14 = (#<CLOSURE (LAMBDA (QT::ACTUAL-INSTANCE QT::ARGS) :IN QT::RESOLVE-CALL) {1001ABC02B}> 8384 :INSTANCE)
  2: ((:METHOD FINALIZE (ABSTRACT-QOBJECT)) #<Qt::QTextEdit::ExtraSelection  {1001E25373}>) [fast-method]
      Locals:
        SB-PCL::.NEXT-METHOD-CALL. = #S(SB-PCL::FAST-METHOD-CALL :FUNCTION #<FUNCTION (SB-PCL::FAST-METHOD FINALIZE (T))> :PV NIL :NEXT-METHOD-CALL NIL :ARG-INFO (1))
        QTOOLS::OBJECT = #<Qt::QTextEdit::ExtraSelection  {1001E25373}>
```

**Comment by Shinmera** [comment]
The former has nothing to do with qtools as far as I know since q+ calls are translated to simple commonqt optimized-calls. I think the problem is the changes stas made to the CFFI parts that I haven't integrated into libcommonqt and rolled out into qt-libs.

As for finalization, same here. It calls maybe-delete-qobject, which calls commonqt's optimized-delete, which is where the error happens.

**Comment by Shinmera** [comment]
I'll deal with the former once I get home.

**Comment by Shinmera** [comment]
With the patch from upstream in shinmera/libcommonqt@619a21fedfdf85826a6c74a3f96ba39c195cf8ab I can confirm it fixed. Getting things rolled out however will take me a while because getting the whole qt-libs chain built on all systems takes a ton of time and is annoying as fuck.

**Comment by Shinmera** [comment]
Should be fixed in shinmera/qt-libs@8a060a6 though I only tested on Linux.

**Comment by phoe** [comment]
Will test it on Windows in due time. Thanks!
